SparseGrids/tsgCoreOneDimensional.hpp

```diff
@@ -184,6 +184,18 @@ namespace OneDimensionalMeta{
     inline bool isTypeCurved(TypeDepth type){ return (getControurType(type) == type_curved); }
 }
 
+//! \internal
+//! \brief Contains algorithms for generating orthogonal polynomials with respect to a NONNEGATIVE weight function.
+//! \ingroup TasmanianCoreOneDimensional
+namespace OneDimensionalOrthPolynomials{
+    //! \brief Generate all roots up to the n roots of the n-th degree orthogonal polynomial.
+    std::vector<std::vector<double>> getRootCache(
+        int n,
+        std::function<double(double)> &weight_fn,
```

Comment:
I would pass in by value here. Otherwise, you may run into issues trying to bind a non-const reference.

Reply:
Changed.